desktop/core/src/desktop/js/reactComponents/FileChooser/PathBrowser/OverflowingItem.tsx

```diff
@@ -0,0 +1,93 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useEffect, useRef, useState } from 'react';
+import { Button } from 'antd';
+
+import OverflowTooltip from './OverflowTooltip';
+import './OverflowingItem.scss';
+
+enum OverflowingComponentType {
+  breadcrumb,
+  menu
+}
+
+type OverflowingComponentTypeString = keyof typeof OverflowingComponentType;
+
+interface OverflowingItemProps {
+  label: string;
+  url: string;
+  handleFilePathChange: (path: string) => void;
+  componentType: OverflowingComponentTypeString;
+}
+
+const customStyles = {
+  breadcrumb: {
+    minWidth: '4ch'
+  },
+  menu: {
+    width: '100%',
+    textAlign: 'left'
+  }
+};
+
+const OverflowingItem: React.FC<OverflowingItemProps> = ({
+  label,
+  url,
+  handleFilePathChange,
+  componentType
+}) => {
+  const textElementRef = useRef<HTMLDivElement>(null);
+  const [isOverflown, setIsOverflown] = useState(false);
+
+  const compareSize = () => {
+    const element = textElementRef.current;
+
+    const compare = element
+      ? element.offsetWidth < element.scrollWidth || element.offsetHeight < element.scrollHeight
+      : false;
+    setIsOverflown(compare);
+  };
+
+  useEffect(() => {
+    compareSize();
+    window.addEventListener('resize', compareSize);
```

Comment:
This is only reacting to resizing of the actual window right? We need to handle the case where a parent component might be resized without the actual browser window resizing (this can happen in Hue when the sidepanels are toggled or resized). Have a look at https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

Reply:
In this scenario, since the pathbrowser is inside a modal I don't think the parent component could be resized as it overlaps.

Reply:
True, but if we aim at creating a reusable hook for this functionality I think it is better to properly handle the resizing.